pdr_backend/analytics/predictoor_stats.py

```diff
@@ -173,31 +172,68 @@ def get_cli_statistics(all_predictions: List[Prediction]) -> None:
         print("No correct predictions found.")
         return
 
-    print(f"Overall Accuracy: {correct_predictions/total_predictions*100:.2f}%")
-
+    data = []
     for key, stat_pair_timeframe_item in stats["pair_timeframe"].items():
-        pair, timeframe = key
+        pair, timeframe, source = key
         accuracy = (
             stat_pair_timeframe_item["correct"]
             / stat_pair_timeframe_item["total"]
             * 100
         )
-        print(f"Accuracy for Pair: {pair}, Timeframe: {timeframe}: {accuracy:.2f}%")
-        print(f"Total stake: {stat_pair_timeframe_item['stake']}")
-        print(f"Total payout: {stat_pair_timeframe_item['payout']}")
-        print(f"Number of predictions: {stat_pair_timeframe_item['total']}\n")
+        new_data = {
+            "timeframe": timeframe,
+            "pair": pair,
+            "source": source,
+            "accuracy": accuracy,
+            "sum_stake": stat_pair_timeframe_item['stake'],
+            "sum_payout": stat_pair_timeframe_item['payout'],
+            "n_predictions": stat_pair_timeframe_item["total"]
+        }
+        data.append(new_data)
 
-    for predictoor_addr, stat_predictoor_item in stats["predictor"].items():
-        accuracy = stat_predictoor_item["correct"] / stat_predictoor_item["total"] * 100
-        print(f"Accuracy for Predictoor Address: {predictoor_addr}: {accuracy:.2f}%")
-        print(f"Stake: {stat_predictoor_item['stake']}")
-        print(f"Payout: {stat_predictoor_item['payout']}")
-        print(f"Number of predictions: {stat_predictoor_item['total']}")
-        print("Details of Predictions:")
-        for detail in stat_predictoor_item["details"]:
-            print(f"Pair: {detail[0]}, Timeframe: {detail[1]}, Source: {detail[2]}")
-        print("\n")
+    df = pl.DataFrame(data, schema=feed_summary_df_schema)
+    return df
+
+
+
+@enforce_types
+def get_predictoor_summary_stats(all_predictions: List[Prediction]) -> pl.DataFrame:
```

Comment:
Rather than doing this, I'd like to see all transform logic using polars dataframes.
Here is generally what this would look like.

1. look at table_pdr_predictions.py _object_list_to_df)
```
predictions_df = _object_list_to_df(predictions, predictions_schema)
```
2. Then, apply dataframe transforms
```
predictions_df = predictions_df.with_columns([
   (pl.col("correct") / pl.col("total")) * 100).as_alias("accuracy"),
   pl.col("stake").sum().alias("sum_stake"),
   pl.col("payout").sum().alias("sum_payout"),
   pl.col("payout").count().alias("n_predictions"),
])
```
3. enforce schema + return values
```
df = pl.DataFrame(predictions_df, schema=predictoor_summary_df_schema)
return df
```